app/models/jobsModel.js

```diff
@@ -1,22 +1,40 @@
 const db = require('../database/dbConfig.js')
+var crypto = require('crypto')
 
 /**
  * Inserts a job into the database.
  * scheduled_time is an optional field.
  * 
  * @param {"class_id" : integer, 
- *         "scheduled_time" : Date, 
- *         "job_hash" : string} job 
+ *         "scheduled_time" : Date} job 
  */
 async function insertJob(job){
     try {
+        job["job_hash"] = crypto.createHash('sha1').update((new Date()).valueOf().toString() + Math.random().toString()).digest('hex')
```

Comment:
1. This a very long line and confusing, it is a good idea to break this up to some different lines
```
const date_string = (new Date()).valueOf().toString();
const random_string = Math.random().toString();
const base_string = date_string+random_string;
job.job_hash = crypto.createHash('sha1').update(base_string).digest('hex');
```
2. use job.job_hash instead of the python key-value pair notation
3. we want to keep the model function as simple as possible, and we want to make sure the parameter passed to these functions are already in a desirable form so that we don't have to modify the parameter anymore. Therefore, this line should be moved to somewhere before this function gets called.